library/std/src/sys/unix/net.rs

```diff
@@ -18,6 +20,86 @@ pub extern crate libc as netc;
 
 pub type wrlen_t = size_t;
 
+pub struct AncillaryDataIter<'a, T> {
+    data: &'a [u8],
+    phantom: crate::marker::PhantomData<T>,
+}
+
+impl<'a, T> AncillaryDataIter<'a, T> {
+    pub fn new(data: &'a [u8]) -> AncillaryDataIter<'a, T> {
+        AncillaryDataIter { data, phantom: PhantomData }
+    }
+}
+
+impl<'a, T> Iterator for AncillaryDataIter<'a, T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        unsafe {
+            let mut unit = mem::zeroed();
+            if mem::size_of::<T>() <= self.data.len() {
+                let unit_ptr: *mut T = &mut unit;
+                libc::memcpy(unit_ptr.cast(), self.data.as_ptr().cast(), mem::size_of::<T>());
```

Comment:
It seems like this is basically `ptr::read`, maybe we should just use that? Or maybe read_unaligned, as it looks like there's no checking for alignment here.

Reply:
The data is unaligned, so `read_unaligned` should be used. I did not know there is a `read_unaligned` method. Thanks for the information.